storage_controller/src/compute_hook.rs

```diff
@@ -71,6 +71,37 @@ impl ComputeHookTenant {
         }
     }
 
+    fn is_sharded(&self) -> bool {
+        matches!(self, ComputeHookTenant::Sharded(_))
+    }
+
+    /// Clear compute hook state for the specified shard.
+    /// Only valid for [`ComputeHookTenant::Sharded`] instances.
+    fn handle_detach(&mut self, tenant_shard_id: TenantShardId, stripe_size: ShardStripeSize) {
```

Comment:
this seems like an overly generic name; if we just want to delete a shard, why not make this `remove_shard`?

the structure this is being called on handles deduplicating of the notices, so one could imagine "handle_detach" could send a message out, but is not of course there.

Reply:
Yeah, I kind of wrote the code with the idea that we'll notify cplane about detaches at some point, but that's not certain. Renamed in https://github.com/neondatabase/neon/pull/9045/commits/2bf618e252de99a1a35b5e6a34e2a3f826d231ec